Code/Framework/AzFramework/Platform/Common/Xcb/azframework_xcb_files.cmake

```diff
@@ -6,6 +6,43 @@
 #
 #
 
+# If not set explicitly, look for a splash image in the project directory
+if(NOT DEFINED SPLASH_FILE)
+    # start by looking for file in the project directory
+    if(EXISTS ${PROJECT_SOURCE_DIR}/Resources/Splash.png)
+        set(SPLASH_FILE ${PROJECT_SOURCE_DIR}/Resources/Splash.png)
+    endif()
+    # if not found, assume it's engine centric build
+    if(DEFINED LY_PROJECTS)
+        if(EXISTS ${LY_PROJECTS}/Resources/Splash.png)
+            set(SPLASH_FILE ${LY_PROJECTS}/Resources/Splash.png)
+        endif()
+    endif ()
+endif()
+# If value is set, check if it exists and generate the header file
+if(DEFINED SPLASH_FILE)
+    file(TO_CMAKE_PATH ${SPLASH_FILE} SPLASH_FILE_RESOLVED_PATH)
+    # check if it's a png file
+    if(NOT ${SPLASH_FILE_RESOLVED_PATH} MATCHES ".*\\.png")
+        message(WARNING "Splash file must be a PNG image, omitting splash screen generation.")
+    else()
+        if(EXISTS ${SPLASH_FILE_RESOLVED_PATH})
+            execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/AzFramework)
+            execute_process(COMMAND ${CMAKE_COMMAND} -E copy ${SPLASH_FILE_RESOLVED_PATH} ${CMAKE_BINARY_DIR}/AzFramework/Splash.png)
```

Comment:
Why does splash file need to be copied to a new folder?

Reply:
It was for organization but I can remove this behaviour

Reply:
This is still copying the file to the ${CMAKE_BINARY_DIR}. Is a copy really needed? `xxd` should be able to use the absolute `${SPLASH_FILE_RESOLVED_PATH}` path instead of needing to make this copy.

Reply:
xdd encodes path to file into the variable name, and for this file generation I need path / filename to be the same when passed to xdd